pkg/controller/postgressnapshot/postgressnapshot_controller.go

```diff
@@ -237,3 +245,52 @@ func (r *ReconcilePostgresSnapshot) createSnapshot(ctx context.Context, rdsSvc r
 	r.logger.Info(msg)
 	return croType.PhaseInProgress, croType.StatusMessage(msg), nil
 }
+
+func buildPostgresSnapshotInfoMetricLabels(cr *integreatlyv1alpha1.PostgresSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := buildPostgresSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
+	if len(string(cr.Status.Phase)) != 0 {
+		labels["statusPhase"] = string(cr.Status.Phase)
+		return labels
+	}
+	return labels
+}
+
+func buildPostgresSnapshotGenericMetricLabels(cr *integreatlyv1alpha1.PostgresSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := map[string]string{}
+	labels["clusterID"] = clusterID
+	labels["resourceID"] = cr.Name
+	labels["namespace"] = cr.Namespace
+	labels["instanceID"] = snapshotName
+	labels["productName"] = cr.Labels["productName"]
+	labels["strategy"] = postgresProviderName
+	labels["statusPhase"] = ""
+	return labels
+}
+
+func (r *ReconcilePostgresSnapshot) exposePostgresSnapshotMetrics(ctx context.Context, cr *integreatlyv1alpha1.PostgresSnapshot) {
+	// build instance name
+	snapshotName := cr.Status.SnapshotID
+
+	// get Cluster Id
+	logrus.Info("setting postgres snapshot information metric")
+	clusterID, err := resources.GetClusterID(ctx, r.client)
+	if err != nil {
+		logrus.Errorf("failed to get cluster id while exposing information metric for %v", snapshotName)
+		return
+	}
+
+	// build metric labels
+	infoLabels := buildPostgresSnapshotInfoMetricLabels(cr, clusterID, snapshotName)
+	// build generic labels
+	genericLabels := buildPostgresSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
```

Comment:
Do we need both of these functions? I think these could probably get merged into one since we don't have 2 separate metrics to label

Reply:
we have 2 separate metrics

Reply:
your right only one at this point, and will refactor to status metric.